src/main/java/com/microsoft/cse/helium/app/services/keyvault/EnvironmentReader.java

```diff
@@ -0,0 +1,166 @@
+package com.microsoft.cse.helium.app.services.keyvault;
+
+import com.microsoft.cse.helium.app.Constants;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import java.util.Arrays;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.core.env.SimpleCommandLinePropertySource;
+import org.springframework.stereotype.Service;
+
+/**
+ * EnvironmentReader
+ * A class to read the environment variables and validate them.
+ */
+@Service
+public class EnvironmentReader implements IEnvironmentReader {
+
+  private String authType;
+  private String keyVaultName;
+  private final String keyVaultNameRegex = "^[a-zA-Z](?!.*--)([a-zA-Z0-9-]*[a-zA-Z0-9])?$";
+  private static final Logger logger = LoggerFactory.getLogger(EnvironmentReader.class);
+
+  /**
+   * EnvironmentReader
+   * This constructor initializes the class with the arguments provided from the CLI
+   * during the Springboot construction of the beans.
+   *
+   * @param applicationArguments the command line arguments.
+   */
+  @SuppressFBWarnings("DM_EXIT")
+  @Autowired
+  public EnvironmentReader(ApplicationArguments applicationArguments) {
```

Comment:
parameters should be --keyvault-name and --auth-type

--auth-type can be MSI (default) CLI or VS

Reply:
This is what I followed as it was linked in the issue tracker. It says "kvname" for key vault name
https://github.com/retaildevcrews/helium-java/issues/122

Same for csharp: 
https://github.com/retaildevcrews/helium-csharp/issues/58

Also, that is how parameters are read. The AppArguments class automatically parses things out and removes the preceding "--"